provider/cmd/pulumi-resource-cloud-toolkit-aws/serverless/queue.ts

```diff
@@ -0,0 +1,156 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import { QueueArgs, defaultConfig, deadLetterQueueTypes } from "./config";
+import { UnsupportedTypeError } from "./errors";
+import { Output } from "@pulumi/pulumi";
+
+export { QueueArgs };
+
+/**
+ * Type Name for the component
+ */
+export const TYPENAME = "cloud-toolkit-aws:serverless:Queue";
+
+const PERMISSIVE_MAX_RECEIVED_COUNT = 10;
+const RESTRICTIVE_MAX_RECEIVED_COUNT = 1;
+const DEFAULT_DL_TYPE = deadLetterQueueTypes.PERMISSIVE;
+const MAX_RECEIVED_COUNT = new Map<deadLetterQueueTypes, number>([
+  [deadLetterQueueTypes.PERMISSIVE, PERMISSIVE_MAX_RECEIVED_COUNT],
+  [deadLetterQueueTypes.RESTRICTIVE, RESTRICTIVE_MAX_RECEIVED_COUNT],
+]);
+
+const STANDARD_SUFFIX = "-standard";
+// Unlike the standard suffix, this is needed in AWS.
+const FIFO_SUFFIX = ".fifo";
+
+// Temporal measure - Should be defined by the user in a future common advanced settings
+const uniqueName = true;
+
+/**
+ * Cloud Toolkit component for Queues. Creates a Simple Queue Service Queue alongside a Dead Letter Queue for faulty message deliveries.
+ */
+export class Queue extends pulumi.ComponentResource {
+  name: string;
+  /** 
+   * Simple Queue Service Queue underline the component.
+   */
+  public readonly sqsQueue: aws.sqs.Queue;
+  /**
+   * Dead Letter Queue associated with the component. Messages that were not delivered will be sent here.
+   */
+  private readonly deadLetterQueue?: string | Output<string>;
```

Comment:
This could be just `Output<string>`

Reply:
I created a public getter alongside 2 private variables to cover the possible cases for the dead letter queue.